vitals/Main.java

```diff
@@ -1,23 +1,64 @@
 package vitals;
 
 public class Main {
-    static boolean batteryIsOk(float temperature, float soc, float chargeRate) {
-        if(temperature < 0 || temperature > 45) {
-            System.out.println("Temperature is out of range!");
-            return false;
-        } else if(soc < 20 || soc > 80) {
-            System.out.println("State of Charge is out of range!");
-            return false;
-        } else if(chargeRate > 0.8) {
-            System.out.println("Charge Rate is out of range!");
-            return false;
-        }
-        return true;
+    
+  public static boolean isTemperatureOk(float temp) {
+    if(temp<0) {
+      System.out.println("Temperature is too low");
+      return false;
     }
+    else if(temp>45) {
+      System.out.println("Temperature is too high");
+      return false;
+    }
+    System.out.println("Temperature is OK");
+    return  true;
+  }
+  
+  public static boolean isChargingStateOk(float soc) {
+    if(soc<20) {
+      System.out.println("charging state is at very low for now, Dont remove charging");
+      return false;
+    }
+    else if(soc>80) {
+      System.out.println("charging state is at High, we can remove charging");
+      return false;
+    }
+    System.out.println("Vehicle is not charged yet, Keep charging ON");
+    return  true;
+  }
+  
+  
+  public static boolean isChargingRateOk(float chargeRate) {
+    if(chargeRate> 0.8) {
+      System.out.println("charging rate is high, Stop charging");
+      return false;
+    }
+    System.out.println("Charging Rate is OK");
+    return  true;
+  }
+  
+  public static boolean isVoltageOk(float volt) {
+    if(volt<2.3) {
+      System.out.println("voltage is too low, kindly stop charging");
+      return false;
+    }
+    else if(volt>4.3) {
+      System.out.println("voltage is too high, kindly stop the charging");
+      return false;
+    }
+    System.out.println("Voltage is OK");
+    return  true;
+  }
 
     public static void main(String[] args) {
-        assert(batteryIsOk(25, 70, 0.7f) == true);
-        assert(batteryIsOk(50, 85, 0.0f) == false);
-        System.out.println("Some more tests needed");
+      testBatteryCharge();
```

Comment:
The method BatteryIsOk() is a single API that provides the status which is a function of all parameters.
By not having such method, the user is expected to know each parameter and verify each parameter separately. Please see if not having that method is convenient or not.